### ISSUE TITLE
Does this implementation supports DualOctreeGNN

### ISSUE THREAD

**Issue by fishfishson** [octree-nn/ocnn-pytorch#12]
Hi author,
Thanks for your pure torch implementation! Your recent work DualOctreeGNN uses ocnn version from microsoft/O-CNN. Can I replace it with this repo?

**Comment by wang-ps** [comment]
Thanks for your interest in our papers. It is definitely possible to replace the microsoft/O-CNN with this ocnn-pytorch since ocnn-pytorch contains all the functionalities of microsoft/O-CNN and more.

And when I am free in the near future, I will also try to do it.

**Comment by fishfishson** [comment]
Looking forward to your next version! BTW if with this repo, can I use the original pre-trained model weight?

**Comment by wang-ps** [comment]
Yes, probably. 
DualOctreeGNN mainly reuses the octree data structure of microsoft/O-CNN, and the convolution is a newly-proposed graph convolution, which may not be affected by the replacement of ocnn-pytorch.

**Comment by fishfishson** [comment]
Thanks! One more question. There exists another great work [Neural Dual Contouring](https://arxiv.org/abs/2202.01999) which applies dual contour to reconstruct surfaces from points. However, it seems like we can add hierarchy ocnn to it for efficiency and accuracy. What do you think of this feasibility?

**Comment by wang-ps** [comment]
Yes, it is a good idea! And I think it is feasible to combine these two works. 
The original version of Dual Contouring uses Octree to speed up. And DualOctreeGNN uses a naive CPU-version marching cube to extract surfaces, which is actually the bottleneck of DualOctreeGNN.

**Comment by fishfishson** [comment]
Thx! Really appreciate all your kind answers.
